grails-app/views/dashboard/megamenu.gsp

```diff
@@ -243,7 +252,28 @@
         </g:hasRoleInvoice>
     </g:if>
 
+    <g:if test="${megamenuConfig.stockRequest.enabled && User.get(session?.user?.id)?.getHighestRole(session.warehouse).roleType == RoleType.ROLE_AUTHENTICATED}">
+        <li class="mm-item">
+            <a href="javascript:void(0)" class="mm-item-link">
+                <warehouse:message code="default.inbound.label" />
+            </a>
+            <div class="mm-item-content">
+                <div class="mm-menu-item">
+                    <g:link controller="stockMovement" action="createRequest">
+                        <warehouse:message code="default.create.label" args="[warehouse.message(code: 'stockRequest.label', default: 'Stock Request')]"/>
+                    </g:link>
+                </div>
+                <div class="mm-menu-item">
+                    <g:link controller="stockMovement" action="list" params="[direction:'INBOUND']">
+                        <warehouse:message code="default.list.label" args="[warehouse.message(code: 'stockMovements.inbound.label')]"/>
+                    </g:link>
+                </div>
+            </div>
+        </li>
+    </g:if>
+
     <g:if test="${megamenuConfig.inbound.enabled}">
+        <g:if test="${User.get(session?.user?.id)?.getHighestRole(session.warehouse).roleType != RoleType.ROLE_AUTHENTICATED}">
```

Comment:
Same here. Also what are we doing here ... saying that the user can access this feature if they don't have ROLE_AUTHENTICATED? Could we do something like roleType > ROLE_AUTHENTICATED? 

Although now I'm seeing that SUPERUSER = 0 so perhaps we did this in the wrong direction. 

@awalkowiak @pmuchowski any thoughts?

Reply:
I would like to say here that user can access this feature if he has role higher than Authenticated.

Reply:
@MagdaToczek if role is higher than Authenticated this means `role_type < ROLE_AUTHENTICATED` (because superuser = 0, admin = 1, ..., authenticated = 5).

Reply:
And `getHighestRole` can return `null`, hence you can get error trying access `roleType` on `null` object.